[13.0][MIG] hr

Yasaie opened the pull request:
Migration of `hr`
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

Yasaie committed:
[MIG] hr

Co-Authored-By: Miquel Raïch <miquel.raich@eficent.com>

MiquelRForgeFlow commented on addons/hr/migrations/13.0.1.1/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,90 @@
+NEW ir.actions.act_window: hr.plan_wizard_action
+NEW ir.actions.act_window: hr.res_users_action_my
+DEL ir.actions.act_window: hr.hr_employee_action_subordinate_hierachy
+NEW ir.actions.report: hr.hr_employee_print_badge [renamed from hr_attendance module]
```
you should handle in this module the xmlid rename

Yasaie replied:
It's `renamed from` it could be not exist. Isn't it better done in `renamed to` side?

MiquelRForgeFlow replied:
if not exist, no problem, the xmlid rename don't break in that case

the rename always is done in destiny module. so if the rename is hr_attendance -> hr, then corresponds to hr module have he xmlid rename script.

Yasaie replied:
Thanks, done

MiquelRForgeFlow commented on addons/hr/migrations/13.0.1.1/pre-migration.py:
```diff
@@ -0,0 +1,20 @@
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.rename_fields(env, _field_renames)
+    # Fix image of hr.employee after renaming column to image_1920
```
do it by orm in post-migration because is image.mixin

MiquelRForgeFlow commented on addons/hr/migrations/13.0.1.1/pre-migration.py:
```diff
@@ -0,0 +1,12 @@
+from openupgradelib import openupgrade
+
+_xmlid_renames = [
+    ('hr.hr_employee_print_badge', 'hr.hr_employee_print_badge')
```
```suggestion
    ('hr_attendance.hr_employee_print_badge', 'hr.hr_employee_print_badge')
```

pedrobaeza committed:
[MIG+FIX] hr: Complete migration scripts

- Update module field definition.

  * With this, we avoid dangling fields in case of uninstallation

- Don't let dangling attachments, having duplicated them for field 'image` and `image_1920`.

pedrobaeza commented:
I have optimized the migration scripts with 2 patches.
